ivy/functional/frontends/tensorflow/raw_ops.py

```diff
@@ -570,6 +570,50 @@ def Conv3D(
     )
 
 
+@to_ivy_arrays_and_back
+def Conv2D(
+    *,
+    input,
+    filter,
+    strides,
+    padding,
+    explicit_paddings=[],
+    use_cudnn_on_gpu=False,
+    data_format='NHWC',
+    dilations=[1, 1, 1, 1],
+    name="Conv2D",
+):
+    # ivy.backends.tensorflow expects strides and dilations to be
+    # a single integer value or a list of 2 values whereas the raw op
+    # expects a list of 4 values
+    if data_format == "NHWC":
+        strides = strides[1:-1]
+        dilations = dilations[1:-1]
+    elif data_format == "NCHW":
+        strides = strides[2:]
+        dilations = dilations[2:]
+    # for Conv2D, the explicit_padding is defined as
+    # [[0, 0], [pad_top, pad_bottom], [pad_left, pad_right], [0, 0]]
+    # when the data_format is "NHWC"
+    # for Conv2D, the explicit_paddings is defined as
+    # [[0, 0], [0, 0], [pad_top, pad_bottom], [pad_left, pad_right],
+    # when the data_format is "NCHW"
+    if padding == "EXPLICIT" and data_format == "NHWC":
+        padding = explicit_paddings[1:2]
+    elif padding == "EXPLICIT" and data_format == "NCHW":
+        padding = explicit_paddings[2:3]
+    else:
+        padding = padding
+    return ivy.conv2d(
```

Comment:
I would suggest placing a breakpoint here first. If the test reaches this breakpoint without failing, then place breakpoints at the start of each ivy's conv2d backend. When a breakpoint is reached, continue running line by line until the test fails. This will indicate the specific line that causes the failure which will give information on how to solve the problem.

Reply:
ok will do that now.

Reply:
ivy's conv2D backend, i didn't understand this

Reply:
ivy.conv2d calls the active backend's conv2d function every time you run the code
These backend functions can be found in layers.py in the backends folder. For example torch's backend is here https://github.com/unifyai/ivy/blob/master/ivy/functional/backends/torch/layers.py

Reply:
ok

Reply:
> 

@AnnaTz when I put the breakpoint at the return i`vy.conv2d(..) `function and debug this is the  error thrown

```
/usr/bin/python3 /opt/.pycharm_helpers/pydev/pydevd.py --multiprocess --qt-support=auto --client 172.17.0.1 --port 35911 --file /ivy/ivy/functional/frontends/tensorflow/raw_ops.py 
Connected to pydev debugger (build 223.8617.48)
2023-02-24 12:01:31.603199: I tensorflow/core/util/util.cc:169] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-02-24 12:01:31.607565: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2023-02-24 12:01:31.607588: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/ivy/ivy/__init__.py", line 16, in <module>
    import tensorflow as tf
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/__init__.py", line 473, in <module>
    keras._load()
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/util/lazy_loader.py", line 41, in _load
    module = importlib.import_module(self.__name__)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/ivy/ivy/functional/frontends/tensorflow/keras/__init__.py", line 2, in <module>
    from . import activations
  File "/ivy/ivy/functional/frontends/tensorflow/keras/activations.py", line 2, in <module>
    from ivy.functional.frontends.tensorflow.func_wrapper import to_ivy_arrays_and_back
  File "/ivy/ivy/functional/__init__.py", line 1, in <module>
    from .ivy import experimental
  File "/ivy/ivy/functional/ivy/__init__.py", line 2, in <module>
    from . import activations
python-BaseException  File "/ivy/ivy/functional/ivy/activations.py", line 26, in <module>
    name: Union[str, None], /, *, custom_objects=Union[ivy.Dict, None]
AttributeError: partially initialized module 'ivy' has no attribute 'Dict' (most likely due to a circular import)
```